driver-core/src/main/com/mongodb/internal/tracing/TracingManager.java

```diff
@@ -0,0 +1,241 @@
+import org.bson.BsonDocument;
+
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import static com.mongodb.MongoClientSettings.ENV_OTEL_ENABLED;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.CLIENT_CONNECTION_ID;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.COLLECTION;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.COMMAND_NAME;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.CURSOR_ID;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.NAMESPACE;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.NETWORK_TRANSPORT;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.QUERY_SUMMARY;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.SERVER_ADDRESS;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.SERVER_CONNECTION_ID;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.SERVER_PORT;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.SESSION_ID;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.SYSTEM;
+import static com.mongodb.internal.tracing.MongodbObservation.LowCardinalityKeyNames.TRANSACTION_NUMBER;
+import static java.lang.System.getenv;
+
+/**
+ * Manages tracing spans for MongoDB driver activities.
+ * <p>
+ * This class provides methods to create and manage spans for commands, operations and transactions.
+ * It integrates with a {@link Tracer} to propagate tracing information and record telemetry.
+ * </p>
+ */
+public class TracingManager {
+    /**
+     * A no-op instance of the TracingManager used when tracing is disabled.
+     */
+    public static final TracingManager NO_OP = new TracingManager(ObservationRegistry.NOOP, false);
+    private final Tracer tracer;
+    private final boolean enableCommandPayload;
+
+    /**
+     * Constructs a new TracingManager with the specified observation registry.
+     *
+     * @param observationRegistry The observation registry to use for tracing operations, may be null.
+     * @param enableCommandPayload Whether to enable command payload tracing.
+     */
+    public TracingManager(@Nullable final ObservationRegistry observationRegistry, final boolean enableCommandPayload) {
+        String envOtelInstrumentationEnabled = getenv(ENV_OTEL_ENABLED);
+        boolean enableTracing = true;
+        if (envOtelInstrumentationEnabled != null) {
+            enableTracing = Boolean.parseBoolean(envOtelInstrumentationEnabled);
+        }
+        tracer = (observationRegistry == null) ? Tracer.NO_OP
+                : (enableTracing) ? new MicrometerTracer(observationRegistry, enableCommandPayload)
+                : Tracer.NO_OP;
```

Comment:
Just to reduce cyclomatic complexity by having less branches.
```suggestion
        tracer = enableTracing && observationRegistry != null ?
                new MicrometerTracer(observationRegistry, enableCommandPayload)
                : Tracer.NO_OP;
```